src/libtls/tls_peer.c

```diff
@@ -286,6 +287,28 @@ static status_t process_server_hello(private_tls_peer_t *this,
 	}
 	extensions->destroy(extensions);
 
+	/* downgrade protection (see RFC 8446, section 4.1.3) */
+	memcpy(server_random_end, &this->server_random[24], 8);
```

Comment:
I'd use chunks for this. This one could be
```suggestion
	server_random_end = chunk_create(&this->server_random[24], 8);
```